R-package/R/xgb.train.R

```diff
@@ -390,7 +390,7 @@ xgb.train <- function(params = xgb.params(), data, nrounds, evals = list(),
     call = match.call(),
     params = params
   )
-
+  bst <- xgb.reset.Booster(bst)
```

Comment:
Should also be called at the end of `xgb.cv` when retaining the boosters.

Reply:
Thank you for the catch, will fix it for CV.

Reply:
I can't find where the `bst_folds` is returned in the CV function, could you please help give a pointer?

Reply:
They are returned through the callback `xgb.cb.cv.predict`:
https://github.com/dmlc/xgboost/blob/0500992cccd1a695fe0030184fb84d0a5f3d703b/R-package/R/callbacks.R#L882

Reply:
But the `save_models` is false: https://github.com/dmlc/xgboost/blob/0500992cccd1a695fe0030184fb84d0a5f3d703b/R-package/R/xgb.cv.R#L196 .

Reply:
Yes, that's the default, but the callbacks are all exported so a user can pass the callback directly with non-default arguments.

Reply:
Thank you for pointing that out. I added a reset to the CV function. It's a bit hacky though, since I'm relying on creating side effects using pointers